homeassistant/components/elkm1/__init__.py

```diff
@@ -313,12 +340,23 @@ def _speak_phrase_service(service):
             return
         elk.panel.speak_phrase(service.data["number"])
 
+    def _set_time_service(service):
+        prefix = service.data["prefix"]
+        elk = _find_elk_by_prefix(hass, prefix)
+        if elk is None:
+            _LOGGER.error("No ElkM1 with prefix for set_time: '%s'", prefix)
+            return
+        elk.panel.set_time()
```

Comment:
We need to pass the timezone from HA since the containers are set to UTC time by default.  

You can get the local time time zone with:
`from homeassistant.util import dt as dt_util`
`dt_util.DEFAULT_TIME_ZONE`

https://github.com/gwww/elkm1/blob/main/elkm1_lib/panel.py#L45

Reply:
Looking at the HA code, `dt_util.DEFAULT_TIME_ZONE` always returns UTC timezone. Is that your understanding as well? 

If so, the fix I'm thinking of is to have a flag on the Elk lib called `utc`. When `utc` is true use `gmtime` else use `localtime`. From HA I would call the Elk `set_time` function with `utc` set to `True`.

Does that sound like a good approach?

Thanks!

Reply:
Assuming I've done the code archeology correctly:

During startup or when the zone is selected in the UI, this function in `core.py` is called

```
    def set_time_zone(self, time_zone_str: str) -> None:
        """Help to set the time zone."""
        time_zone = dt_util.get_time_zone(time_zone_str)

        if time_zone:
            self.time_zone = time_zone
            dt_util.set_default_time_zone(time_zone)
        else:
            raise ValueError(f"Received invalid time zone {time_zone_str}")

```

It will call `dt_util.set_default_time_zone` which will change `DEFAULT_TIME_ZONE` to the local timezone.

Reply:
Since `set_time` already takes a `datetime` object and we expect the elk panel to be in the same timezone as the HA install since its local control, calling `set_time` with  `dt_util.now()` seems to be a solution.